torch/_numpy/random.py

```diff
@@ -98,15 +116,26 @@ def normal(loc=0.0, scale=1.0, size=None):
         size = ()
     dtype = _dtypes_impl.default_dtypes().float_dtype
     values = torch.empty(size, dtype=dtype).normal_(loc, scale)
-    return array_or_scalar(values, return_scalar=size is None)
+    return array_or_scalar(values, return_scalar=size == ())
 
 
 @deco_stream
-@normalizer
 def shuffle(x: ArrayLike):
-    perm = torch.randperm(x.shape[0])
-    xp = x[perm]
-    x.copy_(xp)
+    # no @normalizer because we need to shuffle e.g. lists in-place
```

Comment:
Because we don't want to cast lists to tensors*
We should also remove the `ArrayLike` decorator.